### PR TITLE
[SMALLFIX] Add swiftauth as acceptable authentication method for Swift UFS

### PR DESCRIPTION
**Pull request by madanadit**
Authentication using swiftauth was not supported. Dreamobjects requires said method.

### CODE REVIEW

File: underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java [hunk 1]
```diff
@@ -91,17 +91,24 @@ public SwiftUnderFileSystem(AlluxioURI uri, Configuration configuration) {
...
-      // tempauth requires authentication header to be of the form tenant:user.
-      // JOSS however generates header of the form user:tenant.
-      // To resolve this, we switch user with tenant
-      config.setTenantName(configuration.get(Constants.SWIFT_USER_KEY));
-      config.setUsername(configuration.get(Constants.SWIFT_TENANT_KEY));
+      switch (authMethod) {
+        case "keystone":
+          config.setAuthenticationMethod(AuthenticationMethod.KEYSTONE);
+          break;
+        case "swiftauth":
```
Comment (ifcharming) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: nit: Can you please add a comment pointing to the docs of "swiftauth" ?

Seems like I can't find this in http://docs.openstack.org/developer/swift/overview_auth.html

**Comment by ifcharming** [comment]
@madanadit  Thanks for the improvement! Overall looks good. I left one question.

Also, do you mind opening a ticket for this so we can track it in the release notes?

File: underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java [hunk 2]
```diff
@@ -91,17 +91,24 @@ public SwiftUnderFileSystem(AlluxioURI uri, Configuration configuration) {
...
       config.setTenantName(configuration.get(Constants.SWIFT_TENANT_KEY));
-    } else {
-      config.setAuthenticationMethod(AuthenticationMethod.TEMPAUTH);
-      // tempauth requires authentication header to be of the form tenant:user.
-      // JOSS however generates header of the form user:tenant.
-      // To resolve this, we switch user with tenant
-      config.setTenantName(configuration.get(Constants.SWIFT_USER_KEY));
-      config.setUsername(configuration.get(Constants.SWIFT_TENANT_KEY));
+      switch (authMethod) {
+        case "keystone":
```
Comment (jsimsa) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: please define "keystone" and "swiftauth" as a global named constant

**Comment by jsimsa** [comment]
@madanadit thank you for the fix (you beat me to it) ... left one minor comment

**Comment by madanadit** [comment]
Created a new pull request into branch-1.1: https://github.com/Alluxio/alluxio/pull/3603

**Pull request closed without merging**